### DISCUSSION TITLE
getServerSideProps & getStaticProps sometimes dont work, dont known how-to debug

### DISCUSSION THREAD

**Discussion by albjeremias** [vercel/next.js#52078] [Help] [answered]
### Summary

Hei! :)

so I started a previous discussion to understand how-to use the metadata API.
https://github.com/vercel/next.js/discussions/51942

And now I developed the code using the page router instead of the new app router. but it seems that for some pages it works, for some it won't. I already tried many different things like using the getStaticProps, replacing with getServerSideProps and whenever I test Login page works with any of those.. ButtonFile won't work. I test it using a plugin for firefox to disable javascript and check if the title and metadata are generated.

How can I debug this issue?

https://github.com/helpbuttons/helpbuttons/blob/seo/web/src/pages/Login/index.tsx#L20

https://github.com/helpbuttons/helpbuttons/blob/seo/web/src/pages/ButtonFile/%5Bid%5D.tsx#L21

### Additional information

```js
You can check the full project in:

https://github.com/helpbuttons/helpbuttons this is being developed in the SEO branch

there is a demo in:
https://dev.helpbuttons.org
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

The issue is that you have a Loading state, or rather how you are using it.

This is what the server side rendering process yields:

```html
<!DOCTYPE html>
<html lang="en">
    <head>
       <!-- removed for visibility -->
    </head>
    <body>
        <div id="__next">
            <div class="">
                <div class="loading__"></div>
            </div>
        </div>
        <script id="__NEXT_DATA__" type="application/json">
            {"props":{"pageProps":{}},"page":"/ButtonFile/[id]","query":{},"buildId":"Dso7Ch_0PofeDKXAEjkiR","runtimeConfig":{"apiUrl":"/api/"},"nextExport":true,"autoExport":true,"isFallback":false,"locale":"en","locales":["en","es"],"defaultLocale":"en","scriptLoader":[]}
        </script>
    </body>
</html>
```

That's what Bots and other entities see. You have a special check for `login` where you don't do this. 

The fix would be to exclude this page from the `Loading` status.

To be precise, your loading branch skips rendering the page component, altogether and instead renders a Loading component, not the `<Component {...pageProps} />`.

If you need to do some checks for auth, do them in GetServerSideProps.
